server/src/main/java/org/opensearch/action/admin/cluster/hahealth/ClusterHaHealthResponse.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.hahealth;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.hahealth.ClusterHaHealth;
+import org.opensearch.cluster.health.ClusterHealthStatus;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.xcontent.StatusToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Transport response for Cluster HA Health
+ *
+ * @opensearch.internal
+ */
+public class ClusterHaHealthResponse extends ActionResponse implements StatusToXContentObject {
+
+    private static final String CLUSTER_NAME = "clusterName";
+    private static final String STATUS = "domainHealth";
+    private static final String NUMBER_OF_NODES = "totalNodes";
+    private static final String NUMBER_OF_DATA_NODES = "totalDataNodes";
+    private static final String ACTIVE_SHARDS = "totalShards";
+    private static final String UNASSIGNED_SHARDS = "totalUnassignedShards";
+    private static final String INITIALIZING_SHARDS = "totalInitializingShards";
+    private static final String AWARENESS_ATTRIBUTE = "AwarenessAttribute";
```

Comment:
Most of these variables also present in `ClusterHealthResponse` with the only change is awareness attribute health. If it is just about adding one map in regular health check call, shall we just rather add query param in the health check call which would return more verbose health like here rather than build a new API altogether? Something like `?verbose_health`

What do you think?

Reply:
Same as above